analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Sonar/LocksShouldBeReleased.Mutex.cs

```diff
@@ -0,0 +1,140 @@
+﻿using System;
+using System.Threading;
+
+namespace Net6
+{
+    // https://docs.microsoft.com/en-us/dotnet/standard/threading/mutexes
+    internal class NetMutexTest
+    {
+        public static void AcquiredNotReleased(Mutex paramMutex)
+        {
+            // Note that dispose does not release the resource
+
+            using (var m = new Mutex(true, "foo")) // FN, 'true' means it owns the mutex if no exception gets thrown
+            {
+                // do stuff
+            }
+
+            var m0 = new Mutex(initiallyOwned: true, "bar", out var m0WasCreated); // FN
+            m0.Dispose();
+
+            var m1 = new Mutex(false);
+            m1.WaitOne(); // FN
+            m1.Dispose();
+
+            var m2 = new Mutex(false, "qix", out var m2WasCreated);
+            m2.WaitOne(); // FN
+            m2.Dispose();
+
+            var m3 = Mutex.OpenExisting("x");
+            m3.WaitOne(); // FN
+
+            if (Mutex.TryOpenExisting("y", out var m4))
+            {
+                m4.WaitOne(); // FN
+            }
+
+            var m5 = new Mutex(false);
+            var acquired = m5.WaitOne(200, true); // FN, not released on the corect flow
+            if (acquired)
+            {
+                // here it should be released
+            }
+            else
+            {
+                m5.ReleaseMutex(); // this is a programming error
+            }
+
+            var isAcquired = paramMutex.WaitOne(400, false); // FN
+            if (isAcquired)
+            {
+                // not released
+            }
+        }
+
+        public static void UnsupportedWaitAny(Mutex m1, Mutex m2, Mutex m3)
+        {
+            // it is too complex to support this scenario
+            WaitHandle[] handles = new[] { m1, m2, m3 };
+            var index = WaitHandle.WaitAny(handles);
```

Comment:
Let's release him on at least one execution path to make it a FN (that will not be supported)

Reply:
Yeah! That's properly ugly now 😅